<a href="https://colab.research.google.com/github/mohansaidinesh/Machine-Learning/blob/main/Experiment6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###Experiment : 6

Implement k-Nearest Neighbor algorithm to classify the iris data set. Print both correct and
wrong predictions.

In [ ]:
import matplotlib.pyplot as plt
import numpy as np
from sklearn.datasets import load_iris
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import classification_report, confusion_matrix
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler

In [ ]:
#load the dataset
iris=load_iris()

In [ ]:
iris.data[:5]

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2]])

In [ ]:
#labels are 
print(iris.target_names)

['setosa' 'versicolor' 'virginica']


In [ ]:
#split the training and testing data
X_train, X_test, y_train, y_test = train_test_split(iris.data, iris.target, test_size=0.2, random_state=41)

fitting the model

In [ ]:
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(X_train, y_train)

KNeighborsClassifier()

predicting the label

In [ ]:
y_pred = knn.predict(X_test)

In [ ]:
knn.score(X_train, y_train), knn.score(X_test, y_test)

(0.9833333333333333, 0.9666666666666667)

Confusion matrix is :

In [ ]:
confusion_matrix(y_test, y_pred)

array([[ 9,  0,  0],
       [ 0, 10,  1],
       [ 0,  0, 10]])

In [ ]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00         9
           1       1.00      0.91      0.95        11
           2       0.91      1.00      0.95        10

    accuracy                           0.97        30
   macro avg       0.97      0.97      0.97        30
weighted avg       0.97      0.97      0.97        30



**CONCLUSION :** The accuracy of the training data is 0.98 and for the testing data is 0.96